src/components/phonenumber/PhoneNumber.unit.js

```diff
@@ -11,4 +13,42 @@ describe('PhoneNumber Component', () => {
       Harness.testElements(component, 'input[type="text"]', 1);
     });
   });
+
+  it('Should check mask and value in the phone component', (done) => {
+    const formJson =  {
+      components: [{
+          label: 'Phone Number',
+          tableView: true,
+          allowMultipleMasks: true,
+          inputMasks: [{
+            label: 'mask1',
+            mask: 'mask1'
+          }],
+          key: 'phoneNumber',
+          type: 'phoneNumber',
+          input: true
+       }]
+    };
+    const element = document.createElement('div');
+    Formio.createForm(element, formJson)
+      .then(form => {
+        form.setSubmission({
+          data: {
+            phoneNumber: {
+                value: 'mask1',
+                maskName: 'mask1'
+            }
+        },
+        });
+
+        const phoneNumber = form.getComponent('phoneNumber');
+
+        setTimeout(() => {
+          assert.equal(phoneNumber.data.phoneNumber.maskName, phoneNumber.dataValue.maskName, 'Should check the correct mask name');
```

Comment:
What does this test have anything to do with the changes you made above? Shouldn't you be testing for the output of getValueAsString() to match the table output described above?